proxy.go

```diff
@@ -162,12 +162,87 @@ func (rp *reverseProxy) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
 	requestDuration.With(s.labels).Observe(since)
 }
 
+func executeWithRetry(
+	ctx context.Context,
+	s *scope,
+	startTime time.Time,
+	retryNum int,
+	rp func(http.ResponseWriter, *http.Request),
+	rw ResponseWriterWithCode,
+	srw *statResponseWriter,
+	req *http.Request,
+	proxiedResponseDuration *prometheus.SummaryVec,
```

Comment:
you could even go further. 
```suggestion
	monitorDuration func(float64),
```

on the callee side, you'd do:
```
func(duration float64) {
  	proxiedResponseDuration.With(s.labels).Observe(duration)
}
```

Reply:
Done

Reply:
The point of this fix, I suppose: the time is a float64 value, and we want to neglect the prometheus monitor vector here, so we use a lambda function to achieve that. We include the real prometheus vector when we need it.